web/docs/project/env-vars.md

```diff
@@ -217,21 +218,156 @@ We talk about how to define env vars for each deployment option in the [deployme
 
 ## Custom Env Var Validations
 
-TODO: when the Zod validation PRs are merged, describe how users can define their own validations.
+If your code requires some environment variables, you usually want to ensure that they are correctly defined. In Wasp, you can define your environment variables validation by defining a [Zod object schema](https://zod.dev/?id=basic-usage) and telling Wasp to use it.
+
+:::info What is Zod?
+
+[Zod](https://zod.dev/) is a library that lets you define what you expect from your data. For example, you can use Zod to define that a value should be a string that's a valid email address or that a value should be a number between 0 and 100.
```

Comment:
```suggestion
[Zod](https://zod.dev/) is a library that lets you define what you expect from your data. For example, you can use Zod to define that:
- A value should be a string that's a valid email address.
- A value should be a number between 0 and 100.
- etc.
```
Plus, maybe add an explanation why that is helpful (something along the lines of "this is great because it warns us about improperly defined environment variables as soon as possible instead of failing somewhere else down the line").